compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -40,29 +40,45 @@ pub fn struct_error<'tcx>(tcx: TyCtxtAt<'tcx>, msg: &str) -> DiagnosticBuilder<'
     struct_span_err!(tcx.sess, tcx.span, E0080, "{}", msg)
 }
 
+#[cfg(target_arch = "x86_64")]
+static_assert_size!(InterpErrorInfo<'_>, 8);
+
 /// Packages the kind of error we got from the const code interpreter
 /// up with a Rust-level backtrace of where the error occurred.
 /// Thsese should always be constructed by calling `.into()` on
 /// a `InterpError`. In `librustc_mir::interpret`, we have `throw_err_*`
 /// macros for this.
 #[derive(Debug)]
-pub struct InterpErrorInfo<'tcx> {
-    pub kind: InterpError<'tcx>,
+pub struct InterpErrorInfo<'tcx>(Box<InterpErrorInfoInner<'tcx>>);
+
+#[derive(Debug)]
+struct InterpErrorInfoInner<'tcx> {
+    kind: InterpError<'tcx>,
     backtrace: Option<Box<Backtrace>>,
```

Comment:
Since this is already in a `Box` do we need a second `Box` here?

Reply:
Since this will be `None` in most cases (unless debugging the const evaluator), I'd leave this box in in order to not increase the regular size of the `InterpErrorInfoInner`.

Reply:
But if `kind` is used won't `backtrace` be used as well?

Reply:
`backtrace` is only used if `RUSTC_CTFE_BACKTRACE` is set. In all other cases there will be no `Box` allocated`. I'm not sure how large a `Backtrace` struct is, but if it's larger than `8` bytes, for all normal cases we don't waste space, as no `Box` is allocated. It's fine to have arbitrary costs if `RUSTC_CTFE_BACKTRACE` is used, as that is already expensive by itself